crates/router/src/connector/cryptopay/transformers.rs

```diff
@@ -0,0 +1,240 @@
+                "payment method".to_string(),
+            )),
+        }?;
+        Ok(pay_currency)
+    }
+}
+
+// Auth Struct
+pub struct CryptopayAuthType {
+    pub(super) api_key: String,
+    pub(super) api_secret: String,
+}
+
+impl TryFrom<&types::ConnectorAuthType> for CryptopayAuthType {
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(auth_type: &types::ConnectorAuthType) -> Result<Self, Self::Error> {
+        if let types::ConnectorAuthType::BodyKey { api_key, key1 } = auth_type {
+            Ok(Self {
+                api_key: api_key.to_string(),
+                api_secret: key1.to_string(),
+            })
+        } else {
+            Err(errors::ConnectorError::FailedToObtainAuthType.into())
+        }
+    }
+}
+// PaymentsResponse
+#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "lowercase")]
+pub enum CryptopayPaymentStatus {
+    #[default]
+    New,
+    Completed,
+    Unresolved,
+    Refunded,
+    Cancelled,
+}
+
+impl From<CryptopayPaymentStatus> for enums::AttemptStatus {
+    fn from(item: CryptopayPaymentStatus) -> Self {
+        match item {
+            CryptopayPaymentStatus::New => Self::AuthenticationPending,
+            CryptopayPaymentStatus::Completed | CryptopayPaymentStatus::Refunded => Self::Charged,
+            CryptopayPaymentStatus::Cancelled => Self::Failure,
+            CryptopayPaymentStatus::Unresolved => Self::Unresolved,
+        }
+    }
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+pub struct CryptopayPaymentsResponse {
+    data: CryptopayPaymentResponseData,
+}
+
+impl<F, T>
+    TryFrom<types::ResponseRouterData<F, CryptopayPaymentsResponse, T, types::PaymentsResponseData>>
+    for types::RouterData<F, T, types::PaymentsResponseData>
+{
+    type Error = error_stack::Report<errors::ConnectorError>;
+    fn try_from(
+        item: types::ResponseRouterData<
+            F,
+            CryptopayPaymentsResponse,
+            T,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        let redirection_data = item
+            .response
+            .data
+            .hosted_page_url
+            .map(|x| services::RedirectForm::from((x, services::Method::Get)));
+        Ok(Self {
+            status: enums::AttemptStatus::from(item.response.data.status),
+            response: Ok(types::PaymentsResponseData::TransactionResponse {
+                resource_id: types::ResponseId::ConnectorTransactionId(item.response.data.id),
+                redirection_data,
+                mandate_reference: None,
+                connector_metadata: None,
+                network_txn_id: None,
+            }),
+            ..item.data
+        })
+    }
+}
+
+//TODO: Fill the struct with respective fields
+// REFUND :
+// Type definition for RefundRequest
+#[derive(Default, Debug, Serialize)]
+pub struct CryptopayRefundRequest {
```

Comment:
Do we need Refund structs since we don't have implementation for Refund and Refund Sync?

Reply:
Hey, then we need to remove so much code from transformers.rs and cryptopay.rs